spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -807,9 +809,10 @@ private AbstractIntegrationMessageBuilder<?> buildReply(jakarta.jms.Message jmsR
 		}
 	}
 
-	private Object sendAndReceiveWithContainer(Message<?> requestMessage) throws JMSException {
+	private @Nullable Object sendAndReceiveWithContainer(Message<?> requestMessage) throws JMSException {
 		Connection connection = createConnection(); // NOSONAR - closed in ConnectionFactoryUtils.
 		Session session = null;
+		Assert.notNull(this.replyContainer, "'replyContainer' must not be null");
```

Comment:
We don't need to do this here.
We have respective `if..else` in the `handleRequestMessage()`.
This method has to be marked like:
```
@SuppressWarnings("NullAway") // Dataflow analysis limitation
```

See more info in docs: https://docs.spring.io/spring-framework/reference/7.0/core/null-safety.html#_warnings_suppression